docs/src/description/variables.md

```diff
@@ -0,0 +1,44 @@
+# Local Variables and Built-in Variables
+This section describes the syntax for declaring local variables and built-in variables.
+
+## Variables
+
+In AirScript, variables can be declared in `boundary_constraints` and `integrity_constraints` sections and can contain any expression. Variables can be of three types:
+
+- Scalar
+
+```
+let x = a + \$rand[5]
+```
+
+- Vector
+
+```
+let x = [a + b, c + d, stack_inputs[5] * A]
+```
+
+- Matrix
+
+```
+let x = [[a + b, c + d], [stack_inputs[5] * A, stack_inputs[6] * B]]
+```
```

Comment:
It might be good to add a snippet below that says something like:

"The preceding examples assumed the declaration of constants `A` and `B`, trace columns `a`, `b`, `c`, and `d, a `random_values` array called `rand` and a `pub_inputs`  array called `stack_inputs`"

Or maybe we should just provide the earlier part of the air and say something like

"The preceding examples assumed the Air contained the following declarations:
```
trace_cols:
...
pub_inputs:
...
random_values:
...
```

(I didn't fill the ... out, but obviously we should if doing this)

Reply:
I've refactored this a bit to incorporate your suggestion. Please let me know if you don't agree with the change.